### PR TITLE
Add support for CentOS 8, Debian 10

### PR DESCRIPTION
**Pull request by dhoppe**

### CODE REVIEW

**Comment by dhoppe** [comment]
The acceptance tests need to be refactored, because they are written for Vagrant.

**Comment by ekohl** [comment]
You can also ensure the user before hand:
https://github.com/voxpupuli/puppet-wget/blob/f286e42ab4f5c40770706d9bb1e654b5236b74f5/spec/acceptance/wget_system_spec.rb#L31-L33

That part runs as root and can create the user if needed. Should be the easiest fix.

**Comment by dhoppe** [comment]
This module is broken in so many ways and I do not know, why it has not been archived in favour of ```puppet/archive```. 

Even if I remove the ```$strict``` variables, the acceptance tests will fail, because the user Vagrant does not have ```sudo``` permissions to install ```wget```, if necessary. For example at CentOS hosts.

File: manifests/init.pp [hunk 1]
```diff
@@ -11,10 +11,10 @@
 
   if $manage_package {
     if !defined(Package['wget']) {
-      if $::kernel == 'Linux' {
+      if $facts['kernel'] == 'Linux' {
         package { 'wget': ensure => $version }
       }
-      elsif $::kernel == 'FreeBSD' {
+      elsif $facts['kernel'] == 'FreeBSD' {
         if versioncmp($facts['os']['release']['major'], '10') >= 0 {
```
Comment (ekohl) [inline manifests/init.pp:18]: This is the only supported version, so you can simplify this whole block.

File: spec/acceptance/wget_system_spec.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper_acceptance'
 
 describe 'wget' do
-  let(:wget_manifest) { "class { 'wget': }" }
+  let(:wget_manifest) { "user { 'vagrant': ensure => present, managehome => true } -> class { 'wget': }" }
```
Comment (ekohl) [inline spec/acceptance/wget_system_spec.rb:4]: I'd keep this local to the user test since it's not needed as part of the other tests.

**Comment by Dan33l** [comment]
why we consume energy to maintain a deprecated module ?

**Comment by Dan33l** [comment]
IMO we should set this module in archive mode.

**Comment by alexjfisher** [comment]
> IMO we should set this module in archive mode.

It's a dependency of some of our other modules though??? Fix up those first?

**Comment by alexjfisher** [comment]
> > IMO we should set this module in archive mode.
> 
> It's a dependency of some of our other modules though??? Fix up those first?

Maybe not ours... I have `puppet/wget` in my Puppetfile because I used a nexus module.  But maybe we should adopt that? https://github.com/hubspotdevops/puppet-nexus/issues/116#issuecomment-604409122

**Comment by ekohl** [comment]
Going to close this since it's deprecated. See https://github.com/voxpupuli/puppet-wget/issues/119.

**Pull request closed without merging**